android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/service/MapboxCarAppService.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.maps.testapp.auto.service
+
+import android.annotation.SuppressLint
+import android.content.pm.ApplicationInfo
+import androidx.car.app.CarAppService
+import androidx.car.app.R
+import androidx.car.app.Session
+import androidx.car.app.validation.HostValidator
+import com.mapbox.maps.testapp.auto.car.MapSession
+
+/**
+ * Entry point for the templated app.
+ *
+ * CarAppService is the main interface between the app and Android Auto.
+ *
+ * For more details, see the [Android for Cars Library developer guide](https://developer.android.com/training/cars/navigation).
+ */
+class MapboxCarAppService : CarAppService() {
+  @SuppressLint("PrivateResource")
+  override fun createHostValidator(): HostValidator {
+    return if (applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0) {
+      HostValidator.ALLOW_ALL_HOSTS_VALIDATOR
+    } else {
+      HostValidator.Builder(applicationContext)
+        .addAllowedHosts(R.array.hosts_allowlist_sample)
```

Comment:
What are those hosts?

Reply:
These hosts allow list comes from the official Car app library, I guess we can allow all the hosts validator since this is just a demo app.

Reply:
Since this is a dev app for development purposes, removed the addAllowedHosts and always uses `HostValidator.ALLOW_ALL_HOSTS_VALIDATOR`